Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -0,0 +1,264 @@
+    {
+        private List<Type> migrationHandlerTypes = new List<Type>();
+        /// <summary>
+        /// Returns a copy of all loadable implementation types of IMigrationHandler
+        /// </summary>
+        public List<Type> MigrationHandlerTypes => new List<Type>(migrationHandlerTypes); 
+
+        private List<Object> migrationObjects = new List<Object>();
+        /// <summary>
+        /// Returns a copy of all game objects, prefabs and scene assets selected for migration
+        /// </summary>
+        public List<Object> MigrationObjects => new List<Object>(migrationObjects); 
+
+        private dynamic migrationHandlerInstance;
+
+        public MigrationTool()
+        {
+            RefreshAvailableTypes();
+        }
+
+        /// <summary>
+        /// Adds selectedObject to the list of objects to be migrated. Return false if the object is not of type GameObject, or SceneAsset.
+        /// </summary>
+        /// <param name="selectedObject">Unity Object of type GameObject or SceneAsset</param>
+        public bool TryAddObjectForMigration(Object selectedObject)
+        {
+            if (!selectedObject)
+            {
+                Debug.LogError("Selection is Empty. Could not add for migration.");
+                return false;
+            }
+
+            if (selectedObject is GameObject || selectedObject is SceneAsset)
+            {
+                if (!migrationObjects.Contains(selectedObject))
+                {
+                    migrationObjects.Add(selectedObject);
+                    Debug.Log($"{selectedObject.name} object added for migration");
+                }
+                return true;
+            }
+            Debug.LogError("Object must be a GameObject, Prefab or SceneAsset. Could not add object for migration");
+            return false;
+        }
+
+        /// <summary>
+        /// Adds all prefabs and scene assets found on the assets folder to the list of objects to be migrated
+        /// </summary>
+        public void TryAddProjectForMigration()
+        {
+            AddAllAssetsOfTypeForMigration(new Type[] { typeof(GameObject), typeof(SceneAsset) });
+        }
+
+        /// <summary>
+        /// Removes object from the list of objects to migrated
+        /// </summary>
+        /// <param name="selectedObject">Object to be removed</param>
+        public void RemoveObjectForMigration(Object selectedObject)
+        {
+            migrationObjects.Remove(selectedObject);
+        }
+
+        /// <summary>
+        /// Clears list of objects to be migrated
+        /// </summary>
+        public void ClearMigrationList()
+        {
+            migrationObjects.Clear();
+        }
+
+        /// <summary>
+        /// Migrates all objects from list of objects to be migrated using the selected IMigrationHandler implementation. 
+        /// </summary>
+        /// <param name="type">A type that implements IMigrationhandler</param>
+        /// <returns></returns>
```

Comment:
Nit: Delete this line (this should fail validation because it's an equivalent to an empty line - i.e. the generated docs are the same if you have an empty <returns></returns> or no returns.